search-api/search-actors/src/main/java/org/sunbird/actors/SearchActor.java

```diff
@@ -130,6 +130,21 @@ private SearchDTO getSearchDTO(Request request) throws Exception {
             if (filters.containsKey("relatedBoards"))
                 filters.remove("relatedBoards");
 
+            Map<String, Object> secureSettingsFilter = new HashMap<>();
+            for (String key : filters.keySet()) {
+                if (key.contains("secureSettings") && key.startsWith("secureSettings")) {
```

Comment:
Keep these strings as constants.
startsWith is sufficient ? why to have contains too?

Reply:
Done